elastic/kibana
# [Cases] Convert remaining hooks to React Query

## Summary

This PR converts the remaining cases' hooks to React Query. It also adds the `QueryProvider` to the `CasesProvider` to be able to use the hooks when attaching data to a case from outside cases like ML.

Fixes: https://github.com/elastic/kibana/issues/134663

## Testing

- Verify you can add a comment successfully and the new comment markdown editor is empty after submission
- Verify you can add attachments to a new case and an existing case
- Add a file to a case. Verify that the entry in the user activity is being shown
- Verify that you can push a case to an external service
- Verify that you can push a case to an external service when creating a case
- Verify that the loading spinner is being shown when you edit a comment in the actions of the comment
- Verify that the loading spinner is being shown when you edit a title next to the title
- Verify that the loading spinner is being shown when you edit a tag next to the tags section
- Verify that the loading spinner is being shown when you edit severity next to the severity selectable
- Verify that the status and "Sync alerts" is disabled when changing the status
- Verify that the status and "Sync alerts" is disabled when changing the "Sync alerts"
- Verify that the "edit pencil" is changed to a loading spinner when changing assignees
- Verify you can update a comment
- Verify you can update the fields of the case


### Checklist
